controllers/argocd/custommapper.go

```diff
@@ -149,6 +150,23 @@ func (r *ReconcileArgoCD) namespaceResourceMapper(ctx context.Context, o client.
 		result = []reconcile.Request{
 			{NamespacedName: namespacedName},
 		}
+	} else {
+		// If the namespace does not have the expected managed-by label,
+		// iterate through each ArgoCD instance to identify if the observed namespace
+		// matches any configured sourceNamespace pattern. If a match is found,
+		// generate a reconcile request for the instances.
+		if err := r.Client.List(ctx, argocds, &client.ListOptions{}); err != nil {
+			return result
+		}
+		for _, argocd := range argocds.Items {
+			if glob.MatchStringInList(argocd.Spec.SourceNamespaces, namespaceName, false) {
+				namespacedName := client.ObjectKey{
+					Name:      argocd.Name,
+					Namespace: argocd.Namespace,
+				}
+				result = append(result, reconcile.Request{NamespacedName: namespacedName})
+			}
+		}
```

Comment:
@Mangaal , the source namespace managed by ArgoCD uses a different label `argocd.argoproj.io/managed-by-cluster-argocd`. You would need to check for namespaces that have this label. The logic of adding the mapping to the label is [here](https://github.com/ishitasequeira/argocd-operator/blob/c238af601bb59097ae446f9727807e77b259f04b/controllers/argocd/util.go#L1391-L1434) . The documentation for new label can be found [here](https://argocd-operator.readthedocs.io/en/latest/usage/apps-in-any-namespace/)

Reply:
Hi, @ishitasequeira , there won't be an update even coming for a namespace with the label [argocd.argoproj.io/managed-by-cluster-argocd], only the event for newly created ns and ns with label [argocd.argoproj.io/managed-by] will come. it got filtered in namespaceFilterPredicate() func